lib/vault/api/transform.rb

```diff
@@ -0,0 +1,31 @@
+require_relative '../client'
+require_relative '../request'
+
+module Vault
+  class Client
+    # A proxy to the {Transform} methods.
+    # @return [Transform]
+    def transform
+      @transform ||= Transform.new(self)
+    end
+  end
+
+  class Transform < Request
+    def encode(role_name:, value:, **opts)
```

Comment:
How are the options for encode/decode determined in here? I don't see any references to `batch_input` as a field value for instance.

Reply:
They're part of **opts. Basically, if you toss in `batch_input: *` to the encode/decode methods, it will accept those as part of the options hash that gets passed in the body of the POST request to the client. Ruby is automagical like that sometimes...